Allow setting attribute weights per attribute set.

nikunjkotecha opened the pull request.

nikunjhk committed:
Allow setting attribute weights per attribute set.

nikunjkotecha commented:
cc @sdelbosc

malachy-mcconnell commented:
Hi Nikunj, there are some nice ode changes here but we expect attribute weights to by synchronized from the ecommerce system and not set in Drupal. Do you agree? Is this a quick-fix until we can implement synchronisation of attribute weights (sort order)?

nikunjkotecha commented:
Hi @malachy-mcconnell, I checked the code around automated synchronisation of attribute weights. But that still doesn't support multiple attribute (like the old code). I updated the code around extractConfigurableOptions too in the PR and updated logic should work as is. Probably we may want to set default value in config to `attribute_weights: {  }` as not all systems would have size, otherwise everything here works.

Sortable weights is something we will need to define - reason behind that - we may not have all the attributes in taxonomy terms and it is a performance killer which we should avoid till really required in the project.

miromichalicka commented on modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php:
```diff
@@ -203,7 +203,10 @@ public static function configurableAjaxCallback(array $form, FormStateInterface
-    $configurable_form_settings = \Drupal::service('config.factory')->get('acm_sku.configurable_form_settings');
+
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Inject through DI

miromichalicka commented on modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php:
```diff
@@ -219,13 +222,9 @@ public static function configurableAjaxCallback(array $form, FormStateInterface
-
-      $view_builder = \Drupal::entityTypeManager()
-        ->getViewBuilder('acm_sku');
+      $view_builder = \Drupal::entityTypeManager()->getViewBuilder('acm_sku');
```
Inject through DI

miromichalicka commented on modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php:
```diff
@@ -493,7 +490,12 @@ public function deriveProductTree(SKU $sku) {
 
-    $configurable_weights = \Drupal::service('config.factory')->get('acm_sku.configurable_form_settings')->get('attribute_weights');
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Inject through DI

miromichalicka commented on modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php:
```diff
@@ -664,23 +666,25 @@ public function cartName(SKU $sku, array $cart, $asString = FALSE) {
-  protected function extractConfigurableOptions(array $configurable_options) {
+  protected function extractConfigurableOptions($attribute_set, array $configurable_options) {
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Inject through DI

miromichalicka commented on modules/sku/src/Plugin/AcquiaCommerce/SKUType/Simple.php:
```diff
@@ -27,7 +27,8 @@ public function addToCartForm(array $form, FormStateInterface $form_state, SKU $
 
-    $configurable_form_settings = \Drupal::service('config.factory')->get('acm_sku.configurable_form_settings');
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Inject through DI

nikunjkotecha replied:
@miromichalicka I request to be done latest as part of separate ticket, I don't have a way to test this as of now as we are still on 1.0 which doesn't use dependency injection at all

cc @sdelbosc

miromichalicka commented on modules/sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php:
```diff
@@ -45,8 +45,13 @@ public function addToCartForm(array $form, FormStateInterface $form_state, SKU $
-    $configurable_weights = $configurable_form_settings->get('attribute_weights');
+
+    /** @var \Drupal\acm_sku\CartFormHelper $helper */
+    $helper = \Drupal::service('acm_sku.cart_form_helper');
```
Inject through DI

malachy-mcconnell commented:
On hold. Do not merge until the requested Dependency Injection can be added.

nikunjhk committed:
Merge remote-tracking branch 'upstream/8.x-1.x' into attribute_weights_per_attribute_set

Conflicts:
	modules/acm_sku/src/Plugin/AcquiaCommerce/SKUType/Configurable.php

nikunjkotecha commented:
fixed conflicts

nikunjhk committed:
Fix update numbers in acm_sku.

nikunjhk committed:
Merge remote-tracking branch 'upstream/8.x-1.x' into attribute_weights_per_attribute_set

nikunjhk committed:
Fix merge issue for cartformhelper and schema.

nikunjhk committed:
Merge remote-tracking branch 'upstream/8.x-1.x' into attribute_weights_per_attribute_set

Conflicts:
	modules/acm_sku/acm_sku.services.yml
